### PR TITLE
skip promo popup when logging in

### PR DESCRIPTION
**Pull request by matthewwo**
Added a get request which effectively skips the promotion page.

I couldn't seem to make login work by opening a browser and logging in before using this splitter. Instead I've tweaked the login process such that it will skip the promotion page for the user automatically.

### CODE REVIEW

File: attbillsplitter/main.py [hunk 1]
```diff
@@ -141,16 +141,27 @@ def login(self):
...
 
             print('\U0001F6AB  Login failed. Please check your username and '
                   'password and retry. Or something unexpected happened.')
             return False
 
+    def clickSkipPromo(self):
+        reject_promo_link = (
+            'https://www.att.com/olam/rejectPromoUserResponse.myworld',
+        )
+        skip_promo = self.session.get(reject_promo_link, params={'response':'rejected', 'reportActionEvent':'A_LGN_PROMO_DECLINED_SUB'})
```
Comment (brianzq) [inline attbillsplitter/main.py]: can you make this line less than 80 characters? and space after `:`.

File: attbillsplitter/main.py [hunk 2]
```diff
@@ -141,16 +141,27 @@ def login(self):
             return True
         else:
             if 'promo' in login_submit.url.lower():
-                print ('\U0001F534  Popup window detected during login. '
+                reject_status = self.clickSkipPromo()
+
+                if reject_status == False:
```
Comment (brianzq) [inline attbillsplitter/main.py]: `if not reject_status`

File: attbillsplitter/main.py [hunk 3]
```diff
@@ -141,16 +141,27 @@ def login(self):
...
                        '(Sometimes you might have to do this multiple times.')
-                return False
+
+                return reject_status
 
             print('\U0001F6AB  Login failed. Please check your username and '
                   'password and retry. Or something unexpected happened.')
             return False
 
+    def clickSkipPromo(self):
```
Comment (brianzq) [inline attbillsplitter/main.py]: I've been using underscores over camelcase in this project. Would you mind changing it to `click_skip_promo`?

**Review by brianzq — CHANGES_REQUESTED** [review]
Thanks for the PR. the promo popup window has been a big pain point. This is awesome. Requested some changes in formatting. But overall it's good.

**Comment by matthewwo** [comment]
Thanks for making this project possible!!! Made my life much easier should have found this years ago when I started splitting my ATT plan haha.

Will do the formatting once I get home. Btw, I’m not familiar with the python setup script, how would you compile and test?

**Comment by brianzq** [comment]
I'm glad you find this useful. I'm curious how you find about it?
And currently there is no test. If there are more people contributing to this, I might add some tests and set up continuous integration some day.

**Comment by matthewwo** [comment]
I always wanted to write a script similar to this one but procrastinated :/

Then I decided to lookup if there's any luck on GitHub and I found you made this project!! Really appreciate your effort.

**Review by matthewwo — COMMENTED** [review]
Done reformatting.

File: attbillsplitter/main.py [hunk 4]
```diff
@@ -141,16 +141,35 @@ def login(self):
             return True
         else:
             if 'promo' in login_submit.url.lower():
-                print ('\U0001F534  Popup window detected during login. '
+                reject_status = self.clickSkipPromo()
```
Comment (brianzq) [inline attbillsplitter/main.py]: `click_skip_promo()` here

File: attbillsplitter/main.py [hunk 5]
```diff
@@ -141,16 +141,35 @@ def login(self):
...
+
+                if not reject_status:
```
Comment (brianzq) [inline attbillsplitter/main.py]: no need to define `reject_status` above, just do `if not self.click_skip_promo():`

File: attbillsplitter/main.py [hunk 6]
```diff
@@ -141,16 +141,35 @@ def login(self):
...
+                    print ('\U0001F534  Popup window detected during login. '
                        'Please login you account in a browser and click '
                        'through. Log out your account before you retry. '
                        '(Sometimes you might have to do this multiple times.')
-                return False
+
+                return reject_status
```
Comment (brianzq) [inline attbillsplitter/main.py]: we can simply `return True` here as for it to get here, `reject_status` has to be `True`.

**Review by brianzq — CHANGES_REQUESTED** [review]

File: attbillsplitter/main.py (hunk 6 above)
Comment (matthewwo) [reply attbillsplitter/main.py]: Because there's no return after line 148 inside the if statement. Otherwise put a return false at 149?

**Comment by brianzq** [comment]
LGTM. I'll merge it in and upload to pypi.

**Pull request merged**